sympy/assumptions/refine.py

```diff
@@ -240,6 +240,73 @@ def refine_Relational(expr, assumptions):
     return ask(Q.is_true(expr), assumptions)
 
 
+def refine_re(expr, assumptions):
+    """
+    Handler for real part.
+
+    >>> from sympy.assumptions.refine import refine_re
+    >>> from sympy import Q, re
+    >>> from sympy.abc import x
+    >>> refine_re(re(x), Q.real(x))
+    x
+    >>> refine_re(re(x), Q.imaginary(x))
+    0
+    """
+    from sympy.functions.elementary.complexes import re, im
+    arg = expr.args[0]
+    if arg.is_Atom:
+        if ask(Q.real(arg), assumptions):
+            return arg
```

Comment:
This seems like it would be a valid rule even if `is_Atom` is False. Maybe that if can be removed...

Reply:
I thought I changed this, I added the atom condition because for some reason the I was getting the wrong output. I'll remove it when I make the other required changes.

Reply:
I think I added that condition for cases such as, ``` refine(im(x*y), Q.imaginary(x) & Q.real(y)) ```. Without the ``` is_Atom ```, it returns ``` im(x*y) ``` instead of ```im(x)*y```.

Reply:
Is it able to handle the case, when `arg.is_Atom` is `None`. Or does this case even comes into picture?

Reply:
I am not sure.

Reply:
I don't think is_Atom can be none. Normally in sympy there are lower case `is_` conditions and capitalised ones and these are different. For example `x.is_real` queries whether x is real using the assumptions system and returns True/False/None. On the other hand `x.is_Integer` is a shorthand for `isinstance(x, Integer)` which is either True or False. This is usually the case fo `is_Capitalized` attributes.

Reply:
thanks for the  clarification @oscarbenjamin